### DISCUSSION TITLE
Can Rich's print replace normal python print with datastructure aware color formatting

### DISCUSSION THREAD

**Discussion by dineshdharme** [Textualize/rich#3322] [Ideas]
Let's say we have to print a List, then rich can print the whole list in one determined color with faded color shades.
Similarly formatting can be applied to nested lists. Or nested dictionary or etc. Determining the format through reflection.
Or f-string formatting like

`print(f"{names_list=}")`

I mean subsequently supporting all the datastructures that Rich's print can handle out of the box.

**Comment by willmcgugan** [comment]
I'm afraid I don't understand the question. Can you elaborate and post some examples of what you want?

**Reply by dineshdharme** [reply]
Let's say we have a list

[1,2,3,4] then rich can print the list as [  1(blue), 2(light shade of blue), 3(ligher shade of blue), and so on   ]  This will inform the user visually that all of them belong to same list.

Then if we have list of lists 
```
[[1,2,3], [1,2,3]]  = [   
                                 [1,2,3] (blue and ligher shades changing over the list elements), 
                                 [5,6,7] (green and lighter shades changing over the list elements)

                               ]
```                               
Similar representation for dictionary

`{ a : 1, b: [3,4,7] }`    = the keys 'a' and 'b' could have bold formatting and similar values could have different formatting, say italics 

and color choices.                              

This way Rich's print should be able to handle all common python datastructures transparently so that it will be drop in replacement for python's print.
You already do this for json in another module.

Just like we have color themes for shell and editors, in future, all datastructures say panda's dataframe could pass in themed info of how to print a dataframe or a class and Rich would automatically print it.

**Comment by dineshdharme** [comment]
@willmcgugan : Is it already available in some other module? Because I didn't find such an important usecase mentioned on the Readme.
Sorry if I have missed it because of my ignorance or laziness. 
This function can be present under another namespace if you don't want to pollute existing functions.
Is the idea good or out of scope for Rich ? any reply is appreciated.

**Reply by willmcgugan** [reply]
Have a look at pretty.py in the repository. Maybe you could customize that. Suspect it is better as a third party library.

**Reply by dineshdharme** [reply]
No sure why Rich doesn't see value in the idea proposed. It would at the same time be exceptional demonstraction
of Rich's capabilities. Eyeballing datastructures would get easy.
